### PR TITLE
[nlcst] add type definitions for nlcst

### PR DESCRIPTION
**Pull request by wooorm**
This adds a type only package to represent: <https://github.com/syntax-tree/nlcst>.
This was forked off of `xast` and is similar to `hast and `mdast`.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

/cc @ChristianMurphy @remcohaszing

### CODE REVIEW

File: types/nlcst/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    children: WordContent[];
+}
+
+/**
+ * Node representing typographical devices different from characters which
+ * represent sounds (like letters and numerals), white space, or punctuation.
+ *
+ * It can be used in sentence or word nodes.
+ */
+export interface Symbol_ extends Literal {
```
Comment (wooorm) [inline types/nlcst/index.d.ts:140]: The `ban-types` rule warns when using `Symbol`. `Symbol_` isn’t pretty though.
Should the `ban-types` rule be turned off (and if, how?)

Comment (remcohaszing) [reply types/nlcst/index.d.ts:140]: In `tslint.json` add:

```json
{
    "extends": "dtslint/dt.json",
    "rules": {
        "ban-types": false
    }
}
```

Comment (wooorm) [reply types/nlcst/index.d.ts]: applied!

File: types/nlcst/tslint.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "ban-types": false
+    }
+}
```
Comment (remcohaszing) [inline types/nlcst/tslint.json:6]: Preferably add a newline here, but this isn’t blocking as far as I’m concerned.

**Review by remcohaszing — APPROVED** [review]

File: types/nlcst/tslint.json (hunk 2 above)
Comment (wooorm) [inline types/nlcst/tslint.json:6]:
```suggestion
}
```

Comment (wooorm) [reply types/nlcst/tslint.json]: nice catch! I used `github.dev` but missed that this was missing!

**Review by remcohaszing — APPROVED** [review]

**Review by ChristianMurphy — APPROVED** [review]

File: types/nlcst/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
+// Type definitions for non-npm package nlcst 1.0
+// Project: https://github.com/syntax-tree/nlcst
+// Definitions by: Titus Wormer <https://github.com/wooorm>
+//                 Christian Murphy <https://github.com/ChristianMurphy>
+//                 Remco Haszing <https://github.com/remcohaszing>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
```
Comment (peterblazejewicz) [inline types/nlcst/index.d.ts:7]:
```suggestion
```
It's no-op now, 3.7 (or newer) is only enforced on DT now

**Review by peterblazejewicz — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by wooorm** [comment]
Ready to merge

**Pull request merged**